processing/src/main/java/io/druid/segment/SegmentDesc.java

```diff
@@ -44,51 +49,78 @@ public Interval apply(String identifier)
   // ignores shard spec
   public static SegmentDesc valueOf(final String identifier)
   {
-    String[] splits = identifier.split(DataSegment.delimiter);
-    if (splits.length < 4) {
+    SegmentDesc segmentDesc = parse(identifier);
+    if (segmentDesc == null) {
       throw new IllegalArgumentException("Invalid identifier " + identifier);
     }
-    String datasource = splits[0];
-    DateTime start = new DateTime(splits[1]);
-    DateTime end = new DateTime(splits[2]);
-    String version = splits[3];
+    return segmentDesc;
+  }
+
+  private static SegmentDesc parse(String identifier)
+  {
+    String[] splits = identifier.split(DataSegment.delimiter);
+    if (splits.length < 3) {
+      return null;
+    }
+
+    DateTimeFormatter formatter = ISODateTimeFormat.dateTime();
+
+    String datasource = null;
+    DateTime start = null;
+    DateTime end = null;
+
+    int i = 1;
+    for (; i < splits.length && end == null; i++) {
+      try {
+        DateTime dateTime = formatter.parseDateTime(splits[i]);
```

Comment:
this will still fail in case someone adds creation time to datasource name. e.g ds_{creation_time}
I think one way to solve this is to modify the method and pass in datasource name along with the segment identifiers